src/control/system/database.go

```diff
@@ -724,6 +725,103 @@ func (db *Database) FaultDomainTree() *FaultDomainTree {
 	return db.data.Members.FaultDomains.Copy()
 }
 
+// CompressedFaultDomainTree returns the tree of fault domains of joined
```

Comment:
Hmm. From an organizational standpoint, I'm not sure how I feel about putting these methods on the Database struct vs the Membership struct. The distinction is admittedly a little blurry, but I was trying to keep Database focused on data storage and retrieval, and doing more of the "business logic" above it (e.g. in Membership, etc).

Reply:
Makes sense, I can move it.